packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -724,7 +724,17 @@ export class Project extends ProjectBase {
     const kmsIamResources = new Set<string>();
 
     for (const [name, envVariable] of Object.entries(environmentVariables)) {
-      const envVariableValue = envVariable.value?.toString();
+      let envVariableValue: string;
+      if (SecretValue.isSecretValue(envVariable.value)) {
+        const secretArn = (envVariable.value as SecretValue).secretArn;
+        if (secretArn) {
+          envVariableValue = secretArn;
+        } else {
+          throw new Error('When providing a SecretValue it must resolve to a value from SecretsManager!');
```

Comment:
This would actually be a backwards-incompatible change. If a given `SecretValue`'s `secretArn` is not defined, then we should keep using the `SecretValue` directly.

Please add a unit test that confirms this behavior.

Reply:
Not sure if I get this.
1. Passing of `SecretValues` was not supported prior to these changes right? So were does the incompatibility come from?
2. The changes from above make this irrelevant as well since now the `secretId` is always present
3. The check is meant to verify that the SecretValue actually represents a `SecretValues` from SecretsManager since we do not support `SecretValues` created by `SecretValue.plainText` or `SecretValue.ssmSecure`.

Am I missing something here?

Reply:
> Passing of SecretValues was not supported prior to these changes right? So were does the incompatibility come from?

It was supported - simply by calling `.toString()` on the passed value [here](https://github.com/aws/aws-cdk/blob/4439763a0574d2007b1ef64779ecfdcf1e01c9fd/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L727-L727).

See example test [here](https://github.com/aws/aws-cdk/blob/4439763a0574d2007b1ef64779ecfdcf1e01c9fd/packages/%40aws-cdk/aws-codebuild/test/test.project.ts#L1489-L1505). (In fact, I believe the build is failing exactly for that reason - an existing test is failing)

Reply:
Okay I see what you mean. Though point 2 and 3 still stand in my opinion. 
>The changes from above make this irrelevant as well since now the `secretId` and `secretQualifier` is always present for SecretValues from SecretsManager
The check is meant to verify that the SecretValue actually represents a `SecretValues` from SecretsManager since we do not support `SecretValues` created by `SecretValue.plainText` or `SecretValue.ssmSecure`.
How do you feel about these points?

Regarding the test you mentioned. It actually passes as it should. I managed to break lerna when introducing a dev dependency into `core` for testing purpose. Will have a look to fix it.

Reply:
I feel like my original comment should stand: if the `SecretValue` has the `secretQualifier` property as `undefined`, the behavior should be exactly what it is today (no exception should be thrown).

Reply:
Can you elaborate on this? What would happen if somebody passes:

``` TS
new codebuild.PipelineProject(stack, 'Project', {
  environmentVariables: {
    'ENV_VAR1': {
      type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
      value: cdk.SecretValue.plainText('secret'),
    },
  },
});
```

I am probably not seeing something here. I feel like we would run into an error if a users tries to do this.

Reply:
I agree that it's a weird thing to do, but I don't see much point in not allowing it, especially considering it works today.